BiometricLocker/BiometricLocker.swift

```diff
@@ -80,16 +99,35 @@ final public class BiometricLocker {
         return false
     }
 
-    public enum When {
-        case now
-        case afterTimeAllowance
-        case afterTimeInterval(TimeInterval)
+    /// Store the NotificationCenter observers for deallocation.
+    private var notificationObservers = [NSObjectProtocol]()
+
+    public init() {
+        self.notificationObservers.append(NotificationCenter.default.addObserver(forName: .UIApplicationDidEnterBackground, object: nil, queue: .main) { _ in
+            // If app goes into background, we start the clocks to lock the app.
+            self.lock()
+        })
+
+        self.notificationObservers.append(NotificationCenter.default.addObserver(forName: .UIApplicationWillTerminate, object: nil, queue: .main) { _ in
+            // If the app is killed, lock it instantly.
+            self.lock(.now)
+        })
     }
 
-    /// Locks the app.
-    ///
-    /// - Parameter date: Tells the locker when the app was sent to the background (or any other situation, when applicable), so that we can calculate if the app should be locked.
-    public func lock(_ when: When = .afterTimeAllowance) {
+    deinit {
+        self.notificationObservers.forEach { observer in
+            NotificationCenter.default.removeObserver(observer)
+        }
+    }
+
+    /**
+     Tells the biometric locker to lock the app. Defaults to the currently defined time allowance (5min by default).
+
+     - Important: To change when the app locks after having called this, simply call `lock` again with a different allowance, or tell the app to `lock(.afterTimeInterval(timeInterval)`, as it supercedes the current unlocked time allowance. `unlockedTimeAllowance` is only checked when we ask the BiometricLocker whether the app is locked or not. Changing it before or after calling `lock` should work just fine.
+     - Important:  Keep in mind that changing `unlockedTimeAllowance` after setting a custom time interval will not work as expected.
```

Comment:
It might take more or less time than you'd expect, because we check the time allowance as we "_set the lock_", so if the allowance change, when we check if the app is locked, it will no longer match the custom timeout, but rather the custom timeout ± the difference in time allowance set after the fact. 🤔

Reply:
Even better, make it `{ get }`, avoid any issues. 👍🏽